### PR TITLE
[entropy_src/doc] added main state machine diagram

### PR DESCRIPTION
**Pull request by mwbranstad**
Main state machine figure shows the major state transitions, plus details of the sub-state machines.

Signed-off-by: Mark Branstad <mark.branstad@wdc.com>

### CODE REVIEW

File: hw/ip/entropy_src/doc/_index.md [hunk 1]
```diff
@@ -175,13 +175,7 @@ This drop point will save on conditioner power, and still preserve `esfinal` FIF
...
 The above process will be repeated for as long as entropy bits are to be collected and processed.
 
-At any time, the `ENABLE` field can be cleared to halt the entropy generation (and health check testing) within a few clocks cycles after the register write.
-The reason for the delay is that there is a power spike feature where a handshake exists between ENTROPY_SRC and CSRNG modules.
-The power spike feature prevents the cryptographic block in ENTROPY_SRC to run at the same time as the one in CSRNG.
-A handshake exists that is made up of a simple request/acknowledge interface.
-The ENTROPY_SRC block will make a request to CSRNG.
-The CSRNG block will acknowledge after any current block encrpytion request has completed.
-At that point, the ENTROPY_SRC block will recieve the acknowledge and continue processing entropy bits, running them through the SHA3 conditioner.
+At any time, the `ENABLE` field can be cleared to halt the entropy generation (and health check testing) immediately.
```
Comment (tjaychen) [inline hw/ip/entropy_src/doc/_index.md:178]: just a quick question here, so what happened to the original need to quiet the interfaces?

**Review by tjaychen — APPROVED** [review]

File: hw/ip/entropy_src/doc/_index.md (hunk 1 above)
Comment (mwbranstad) [reply hw/ip/entropy_src/doc/_index.md:178]: Need is still there. Perhaps the word "immediately" (kind of relative) is too strong. Should this be re-written to say "after the entropy_src and csrng blocks have quiesced to an idle state"?

**Comment by tjaychen** [comment]
ah it might be, just to make sure software doesn't toggle it off and then
on immediately and expect everything to be fine.
Unless... doing so would be perfectly legal?

I wouldn't necessarily update the language here, you could also add a note
for it in the programmers model, all would be fine to me.

**Review by cdgori — COMMENTED** [review]
Post-commit, the state machine diagram LGTM.

**Pull request merged**